editor/js/Menubar.Edit.js

```diff
@@ -148,10 +148,89 @@ function MenubarEdit( editor ) {
 
 	} );
 	options.add( option );
-
+	
 	//
 
 	options.add( new UIHorizontalRule() );
+	
+	// Skin
+
+	option = new UIRow();
+	option.setClass( 'option' );
+	option.setTextContent( strings.getKey( 'menubar/edit/skin' ) );
+	option.onClick( function () {
+
+		let object = editor.selected;
+
+		if ( object === null || object.parent === null || !object.isMesh ) return; // avoid cloning the camera or scene
+		
+		var i=0;
+		var bones = [];
+    object.traverse( function( o ) {
+      if ( o.isBone ) {
+        bones[i] = o;
+        i++;
+      }
+    } );
+		const skeleton = new THREE.Skeleton( bones );
+		
+		const positions = object.geometry.attributes.position;
+    const vertex = new THREE.Vector3();
+    const center = new THREE.Vector3();
+        
+    const skinIndices = [];
+    const skinWeights = [];
+        
+    for ( let i = 0; i < positions.count; i ++ ) {
+        
+      vertex.fromBufferAttribute( positions, i );
+      var d = 0;
+      var dist = 0;
+      var indx = [];
+      indx[0] = 0; indx[1] = 0; indx[2] = 0; indx[3] = 0;
+        					
+      for (var j=0; j<skeleton.bones.length; j++) {
+        					  
+        skeleton.bones[ j ].getWorldPosition( center );
+        					  
+        d = Math.abs( center.distanceTo( vertex ) );
+        if ( j == 0 || d < dist ) {
+        					    
+          dist = d;
+        	if ( skeleton.bones[ j ].children.length == 0 ) {
+        		for (var k=0; k<4; k++) {
+        			indx[k] = j;
+        		}
+        		skeleton.bones[ j ].name = 'Endsite';
+        	} else {
+        		indx[0] = j;
+        		for (var k=1; k<4; k++) {
+        			for (var l=0; l<skeleton.bones.length; l++) {
+        				if ( skeleton.bones[ indx[k-1] ].parent && skeleton.bones[ indx[k-1] ].parent == skeleton.bones[ l ] ) {
```

Comment:
I read it three times and still did not understand what it does 😅 does it find the closest bone and sets the weights to its parent chain?

Reply:
Yes, it sets the closest center of bone for major influence and its parents as minor.
It is probably not the best possible function, just the one that I came up with.
The only example provided by three.js involves parameters of a cylinder (https://threejs.org/docs/#api/en/objects/SkinnedMesh). 
Are there any industry-standard solutions for this? :)

Reply:
As mentioned in https://github.com/mrdoob/three.js/issues/26501#issuecomment-1657107116, the `three.js editor` is a pure scene editor. And it's not the feature of a scene editor to author skeletons/armatures or animations. This is usually done in a software like Blender. It's really no good strategy to mix the concepts of a scene editor and a digital content creation tool. Both pieces of software should focus on different things. 

Besides, I'm afraid the PR in its current form won't be much usable in production. Especially the assignment of fixed 
skin weights do not reflect how skinning is used in real world apps.